File: src/main/scala/edu/nus/bd/ingest/RDDLoad.scala
package edu.nus.bd.ingest

import org.apache.log4j.{Level, LogManager}
import org.apache.spark.SparkConf
import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SparkSession

object RDDLoad {
  def main(args: Array[String]): Unit = {

    val spark = SparkSession
      .builder()
      .config(new SparkConf())
      .appName("RDD Load Pipeline")
      .master("local[*]")
      .getOrCreate()

    LogManager.getRootLogger.setLevel(Level.WARN)
    spark.sparkContext.setLogLevel("ERROR")

    val lines: RDD[String] = spark.sparkContext.textFile("/Users/arunma/IdeaProjects/SparkJobLoadShowTemplate/src/main/resources/NameEmailCountry.csv", 3)

    val agesAsStrings = lines.map(line => line.split("\\|")(2))

    val agesAsNumbers = agesAsStrings.map(ageStr => ageStr.toInt)

    val totalAge = agesAsNumbers.reduce((one, two) => one + two)

    print (s"Total Age : $totalAge")

    Thread.sleep(100000)

  }


}

File: pyspark/RDDLoad.py
from pyspark.sql import SparkSession

spark = SparkSession.builder.appName('RDD Load Python').getOrCreate()

sc = spark.sparkContext

lines = sc.textFile("/Users/arunma/IdeaProjects/SparkJobLoadShowTemplate/src/main/resources/NameEmailCountry.txt")

agesAsStrings = lines.map(lambda line: line.split("|")[2])
agesAsNumbers = agesAsStrings.map(lambda str: int(str))
totalAge = agesAsNumbers.reduce(lambda one, two: one + two)

lines.count()
lines.take(2)
lines.collect()
type(lines.collect())


#Creating a PairRDD
agesToNames = lines.map(lambda line: (line.split("|")[2], line.split("|")[0]))

agesToNames.reduceByKey(lambda v1, v2: v1 + "\n" + v2)

#Finding number of people in the same age
total = agesToNames.mapValues (lambda x: 1).reduceByKey(lambda v1, v2: v1 + v2)
total.collect()


print (totalAge)


total.takeOrdered(5, lambda x: x[1])
total.takeOrdered(5, lambda x: -x[1])
